File: ios/MidTrim/MidTrim/Domain/UseCases/DeleteProjectUseCase.swift
import Foundation

struct DeleteProjectUseCase {
    private let projectRepository: ProjectRepositoryProtocol
    private let fileRepository: VideoFileRepositoryProtocol

    init(projectRepository: ProjectRepositoryProtocol, fileRepository: VideoFileRepositoryProtocol) {
        self.projectRepository = projectRepository
        self.fileRepository = fileRepository
    }

    func execute(project: ProjectInfo) async throws {
        try? await fileRepository.deleteOutputVideo(at: project.outputVideoURI)
        try? await fileRepository.deleteThumbnail(at: project.thumbnailURI)
        try await projectRepository.delete(by: project.id)
    }
}

File: android/app/src/main/java/com/izzatismail/midtrim/domain/usecase/DeleteProjectUseCase.kt
package com.izzatismail.midtrim.domain.usecase

import com.izzatismail.midtrim.domain.entity.ProjectInfo
import com.izzatismail.midtrim.domain.repository.ProjectRepository
import com.izzatismail.midtrim.domain.repository.VideoFileRepository

class DeleteProjectUseCase(
    private val projectRepository: ProjectRepository,
    private val fileRepository: VideoFileRepository
) {
    suspend fun execute(project: ProjectInfo) {
        try { fileRepository.deleteOutputVideo(project.outputVideoUri) } catch (_: Exception) {}
        try { fileRepository.deleteThumbnail(project.thumbnailUri) } catch (_: Exception) {}
        projectRepository.delete(project.id)
    }
}